code/datums/status_effects/debuffs.dm

```diff
@@ -81,6 +81,26 @@
 	icon_state = "asleep"
 
 //OTHER DEBUFFS
+/datum/status_effect/noheal //you can't heal brute or burn.
+	id = "noheal"
+	duration = -1
+	tick_interval = 0
+	alert_type = null
+	var/lastbrute
+	var/lastfire
+
+/datum/status_effect/noheal/tick()
+	var/newbrute = owner.getBruteLoss()
+	if(newbrute <= lastbrute)
+		var/brutediff = newbrute - lastbrute
+		owner.adjustBruteLoss(brutediff)
+	lastbrute = newbrute
+	var/newfire = owner.getFireLoss()
+	if(newfire <= lastfire)
+		var/firediff = newfire - lastfire
+		owner.adjustFireLoss(firediff)
+	lastfire = newfire
```

Comment:
Really not sure how I feel about how this works. On the one hand mob code is a bitch and I understand it's a pain to do some things properly in there. On the other hand having a ticking operation to just undo changes elsewhere is disgusting.

Talk to me in coderbus sometime about doing this better. This needs to be changed to prevent healing rather than undoing previous changes.

Reply:
i'm trying out bitflags to see if that's better, one moment